### PR TITLE
feat(cognito): user pool client logout urls

### PR DESCRIPTION
**Pull request by tneely**
Support Logout URLs in UserPoolClient.

This is a pretty minimal change to just allow the property to be set.

fixes: https://github.com/aws/aws-cdk/issues/10225

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by lielran** [comment]
Great stuff! Thanks @tneely

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts [hunk 1]
```diff
@@ -56,6 +56,12 @@ export interface OAuthSettings {
    */
   readonly callbackUrls?: string[];
 
+  /**
+   * List of allowed logout URLs for the identity providers.
+   * @default - no logout URLs
+   */
+  readonly logoutUrls?: string[];
```
Comment (nija-at) [inline packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts:63]: Any reason you've added it within `OAuthSettings`? AFAICT from the documentation, this isn't related to OAuth.

**Review by nija-at — CHANGES_REQUESTED** [review]

**Review by nija-at — CHANGES_REQUESTED** [review]
> I can make this more inline with the callbackUrls behavior, but I'm
not sure what the intent is behind that logic. Is it needed for the CFn
template to deploy correctly, or are we trying to nudge users towards
using the property because without it you won't be able to sign in/out
properly?

I don't recall which one it was, in this case, but likely so that the template
deploys correctly. In general, in the CDK, we try our best to do both - let
users produce a correct template and guide them towards the correct use
of properties so that the Cognito app is set up correctly.

Could you deploy a cdk app with this value set and report back?

File: packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts (hunk 1 above)
Comment (lielran) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts:63]: Callback url and Logout urls are part of the oAuth configuration. 
 
the settings in the UI is under `App client settings` (title "What identity providers and OAuth 2.0 settings should be used for your app clients".

Comment (lielran) [reply packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts:63]: worth mention that CFN user pool clients is flat. but here in CDK I see a structure by `oAuth`. 
anyhow, logoutUrls should be besides callbarckIUrls that currently inside the `oAuth` object.

**Comment by tneely** [comment]
I tried deploying the following resource with and without CallbackURLs and LogoutURLs. The template deploys just fine without LogoutURLs, but gives the following error when CallbackURLs is missing: `CallbackUrls can not be empty when code flow or implicit flow is selected `

```
"AuthenticationUserPoolUserPoolClient8AE1704E": {
      "Type": "AWS::Cognito::UserPoolClient",
      "Properties": {
        "UserPoolId": {
          "Ref": "AuthenticationUserPool28698864"
        },
        "AllowedOAuthFlows": ["implicit", "code"],
        "AllowedOAuthFlowsUserPoolClient": true,
        "AllowedOAuthScopes": [
          "profile",
          "phone",
          "email",
          "openid",
          "aws.cognito.signin.user.admin"
        ],
        "CallbackURLs": ["https://www.example.com"],
        "LogoutURLs": ["https://www.example.com"],
        "SupportedIdentityProviders": ["COGNITO"]
      }
    }
  }
```

**Review by nija-at — APPROVED** [review]

**Pull request merged**